### DISCUSSION TITLE
Setup with libvirt

### DISCUSSION THREAD

**Discussion by sajusat** [k3s-io/k3s#9091] [General]
I am trying to setup a cluster with libvirt provider for Vagrant. Unfortunately libvirt creates a network interface for management and that is the default one. It is using DHCP. I have created a private NAT network in vagrant but K3s control plane node is listening on the default network interface created by libvirt.
Can someone tell me how to do it correctly. Any help would be appreciated.

**Comment by dereknola** [comment]
Use the `--flannel-iface` and `--node-external-ip` flags to configure the connection on the private network. We ran into this with our vagrant CI. See https://github.com/k3s-io/k3s/blob/master/tests/e2e/validatecluster/Vagrantfile#L49-L50. Also you can look into the [vagrant-k3s](https://github.com/k3s-io/vagrant-k3s) plugin if you are interested.
